### PR TITLE
HDK-rust immigration

### PR DESCRIPTION
**Pull request by lucksus**
We decided to move HDK-rust into this repository. 
* merged the API serialization branch 
* took resulting head of master (22e1fc3c950d6f786ac1a5efc585309181f52e90) after and copied all files over to here
* adjusted paths
* removed warnings
* added building of wasm-test to makefile
* ran linter
* fixed url in hdk-rust readme
* added `get_entry` to list of working functions

### CODE REVIEW

File: hdk-rust/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+[![Chat](https://img.shields.io/badge/chat-chat%2eholochain%2enet-blue.svg?style=flat-square)](https://chat.holochain.net)
+
+## Overview
+`hdk-rust` is a library for Rust-based holochain dApps that makes it easier to develop Holochain zomes. With Holochain, zome functions and validation code are represented as WASM binaries. This library provides bindings for Rust.
+
+## Usage
+First, [Rust](https://www.rust-lang.org/en-US/install.html) must be installed on your computer.
+
+Being a Rust library, `hdk-rust` can be added as a dependency to any Rust crate. When you generate Rust based Zomes with [holochain-cmd](https://github.com/holochain/holochain-cmd) it will automatically be added as a dependency, and imported into your code.
+
```
Comment (Connoropolous) [inline hdk-rust/README.md]: There are def some updates to make to the readme, but I'll wait till this goes in and propose them

**Comment by Connoropolous** [comment]
at the moment, tests are still failing so there must be something small in there to sort out

**Comment by ddd-mtl** [comment]
How about renaming it just `hdk` because `\holochain-rust\hdk-rust` is kind of silly no?

Or even better `\holochain-rust\zome-framework`

**Comment by Connoropolous** [comment]
That's making a lot of sense to me Damien

**Comment by ddd-mtl** [comment]
woot

**Pull request merged**